hooks/tk-flame_actions.py

```diff
@@ -142,103 +167,127 @@ def execute_action(self, name, params, sg_publish_data):
         :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
         :returns: No return value expected.
         """
+
         app = self.parent
         app.log_debug("Execute action called for action %s. "
                       "Parameters: %s. Publish Data: %s" % (name, params, sg_publish_data))
 
         try:
-            if name == "load_clip":
+            if name == CLIP_ACTION:
                 self._import_clip(sg_publish_data)
 
-            elif name == "load_setup":
+            elif name == BATCH_ACTION:
                 self._import_batch_file(sg_publish_data)
 
-            elif name == "load_batch":
-                self._import_batch_group_from_shot(sg_publish_data)
+            elif name == SHOT_CREATE_ACTION:
+                self._add_batch_group_from_shot(sg_publish_data, True)
+
+            elif name == SHOT_LOAD_ACTION:
+                self._add_batch_group_from_shot(sg_publish_data, False)
 
             else:
-                raise FlameActionError("Unknown action name")
-        except FlameActionError as error:
-            self.parent.log_warning("Flame Action Error: {}".format(str(error)))
+                raise FlameActionsError("Unknown action name")
+        except FlameActionsError as error:
+            # A FlameActionError reaching here mean that something major have stopped the current action
+            app.log_error(error)
 
     ##############################################################################################################
     # methods called by the menu options in the loader
 
     def _import_batch_file(self, sg_publish_data):
         """
-        Imports a Batch Group from Shotgun into Flame.
+        Imports a Batch setup into Flame.
+
+        This function import the Batch setup into the current Batch Group.
         
         :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
         :type sg_publish_data: dict
         """
 
-        batch_path = self.get_publish_path(sg_publish_data)
+        app = self.parent
+        app.log_debug("Importing batch file using '%s'" % sg_publish_data)
+
+        setup_path = self.get_publish_path(sg_publish_data)
 
-        # Only load the batch if it exists
-        if batch_path and os.path.exists(batch_path):
+        # Directly load the setup if exists
+        if setup_path and os.path.exists(setup_path):
             flame.batch.go_to()
-            flame.batch.load_setup(batch_path)
-            flame.batch.organize()
+
+            if not flame.batch.load_setup(setup_path):
```

Comment:
Does the flame python API return error?

I'm pretty sure the flame code will check for setup existence when opening the file.
Would be faster to avoid stat'ing the setup file at every layer.